users/grants/grantsRouter.js

```diff
@@ -0,0 +1,106 @@
+const router = require("express").Router();
+
+const Grants = require("./grantsModel");
+const restricted = require("../../auth/middleware/restricted");
+const checkUserId = require("../../auth/middleware/verifyUserId");
+
+//get all grants (maybe add a filter)
+router.get("/", restricted, (req, res) => {
+  //should we filter grants by a status?? ie: open vs. closed grants
+  Grants.findGrants()
+    .then((grants) => {
+      res.json(grants);
+    })
+    .catch((err) => res.send(err));
+});
+
+//get a grant by grant_id
+router.get("/:grantId", (req, res) => {
+  const { grantId } = req.params;
+  Grants.findSingleGrantBy(grantId)
+    .then((grant) => {
+      res.status(200).json({
+        grant,
+      });
+    })
+    .catch((err) => {
+      res.status(500).json({
+        message: "there's been an issue getting list of grants",
+        error: err,
+      });
+    });
+});
+
+//put update a grant info
+router.put("/:grantId", restricted, (req, res) => {
+  const { grantId } = req.params;
+  const changes = req.body;
+  Grants.updateGrant(changes, grantId)
+    .then((updated) => {
+      res.status(204).json({
+        recordsUpdated: updated,
+      });
+    })
+    .catch((err) => {
+      res.status(500).json({
+        message: "There was an issue updating grant information",
+        error: err,
+      });
+    });
+});
+
+//delete a grant by grant_id
+router.delete("/:grantId", restricted, (req, res) => {
+  const grantId = req.params.grantId;
+  Grants.deleteGrant(grantId)
+    .then((grant) => {
+      res.status(200).json({
+        message: "The writer grant successfully deleted.",
+      });
+    })
+    .catch((err) => {
+      res.status(500).json({
+        message: "There was an error deleting this grant",
+        error: err,
+      });
+    });
+});
+
+// *** Affecting grants based on user ***
+
+//get all grants
+router.get("/user/:userId/", restricted, (req, res) => {
+  const userId = req.params.userId;
+  //should we filter grants by a status?? ie: open vs. closed grants
```

Comment:
This is a good idea so we can display what grants are available for a writer to assist with or not.